Sources/SwiftSDKGenerator/SystemUtils/Shell.swift

```diff
@@ -130,61 +106,40 @@ final class Shell {
 
   static func readStdout(
     _ command: String,
-    currentDirectory: FilePath? = nil,
     shouldLogCommands: Bool = false,
     file: String = #file,
     line: Int = #line
   ) async throws -> String {
-    let process = try Shell(
-      command,
-      currentDirectory: currentDirectory,
-      shouldDisableIOStreams: false,
-      shouldLogCommands: shouldLogCommands,
-      file: file,
-      line: line
+    if shouldLogCommands {
+      print(command)
+    }
+
+    let result = try await ProcessExecutor.runCollectingOutput(
+      executable: "/bin/sh",
+      ["-c", command],
+      collectStandardOutput: true,
+      collectStandardError: false,
+      perStreamCollectionLimitBytes: 100 * 1024 * 1024
```

Comment:
```suggestion
      perStreamCollectionLimitBytes: 100 * 1024 * 1024,
      environment: ProcessInfo.processInfo.environment
```
We need to inherit the environment here, otherwise the `sh` process will run with a default path of `"/usr/gnu/bin:/usr/local/bin:/bin:/usr/bin:."`.   This is sufficient for many of the commands we call using `readStdout`, such as `xar`, `rsync` and even `swift`, but it will cause `launchDockerCommand` to fail because `docker` is in the Homebrew path.

(The other functions using `docker`, such as `copyFromDockerContainer`, don't have this problem because they use `Shell.run`, which _does_ inherit the environment.)

Reply:
Nice catch, thanks!